cmd/krel/cmd/release_notes.go

```diff
@@ -0,0 +1,191 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/release/pkg/command"
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/notes"
+	"k8s.io/release/pkg/notes/options"
+	"k8s.io/release/pkg/util"
+)
+
+// releaseNotesCmd represents the subcommand for `krel release-notes`
+var releaseNotesCmd = &cobra.Command{
+	Use:   "release-notes",
+	Short: "release-notes is the tool of choice for the Release Notes subteam of SIG Release",
+	Long: fmt.Sprintf(`krel release-notes
+
+The 'release-notes' subcommand of krel has been developed to:
+
+1. Generate the release notes for the provided tag for commits on the master
+   branch. We always use the master branch because a release branch
+   gets fast-forwarded until we hit the first release candidate (rc). This is
+   also the reason why we select the first 'v1.xx.0-rc.1' as start tag for
+   the notes generation.
+
+2. Put the generated notes into a release notes draft markdown document and
+   create a GitHub pull request targeting to update the file:
+   https://github.com/kubernetes/sig-release/blob/master/releases/release-1.xx/release-notes-draft.md
+
+3. Put the generated notes into a JSON file and create a GitHub pull request
+   to update the website https://relnotes.k8s.io.
+
+To use the tool, please set the %v environment variable which needs write
+permissions to your fork of k/sig-release and k-sigs/release-notes.`,
+		options.TokenKey),
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runReleaseNotes()
+	},
+}
+
+type releaseNotesOptions struct {
+	tag string
+}
+
+type releaseNotesResult struct {
+	markdown string
+	json     string
+}
+
+var releaseNotesOpts = &releaseNotesOptions{}
+
+func init() {
+	releaseNotesCmd.PersistentFlags().StringVarP(
+		&releaseNotesOpts.tag,
+		"tag",
+		"t",
+		"",
+		"version tag for the notes",
+	)
+
+	rootCmd.AddCommand(releaseNotesCmd)
+}
+
+func runReleaseNotes() (err error) {
+	var tag string
+	if releaseNotesOpts.tag == "" {
+		tag, err = tryToFindLatestMinorTag()
+		if err != nil {
+			return errors.Wrapf(err, "unable to find latest minor tag")
+		}
+	} else {
+		tag = releaseNotesOpts.tag
+	}
+
+	s, err := util.TagStringToSemver(tag)
+	if err != nil {
+		return errors.Wrapf(err, "no valid tag: %v", tag)
+	}
+	start := util.SemverToTagString(semver.Version{
+		Major: s.Major,
+		Minor: s.Minor - 1,
+		Patch: 0,
+		Pre:   []semver.PRVersion{{VersionStr: "rc.1"}},
+	})
+	logrus.Infof("Using start tag %v", start)
+	logrus.Infof("Using end tag %v", tag)
+
+	_, err = releaseNotesFrom(start)
+	if err != nil {
+		return errors.Wrapf(err, "generating release notes")
+	}
+
+	//TODO: implement PR creation for k-sigs/release-notes and k/sig-release
```

Comment:
ref: https://github.com/kubernetes/release/issues/1061

Reply:
@JamesLaverack do you want to continue with that? :blush:

Reply:
@saschagrunert how were you planning on doing the automation for the PR creation? Using the usual fork → branch → push → PR flow ?

Reply:
Yes, I think this is the easiest way to implement it :)

Reply:
(Apologies, I think I've got something configured to ignore GitHub notifications.)

Agree that a fork approach is best. We're going to need to put a GitHub token in CI for this tool to use during the release process. Without a fork we'd either need to give that token permission to create branches on `k-sigs/release-notes` and `k/sig-release` (😬), or let it push directly to `master` on both of those repositories and avoid the PR altogether (😱).

Reply:
Or we switch to a bucket here as well, but this would make the results of the periodical release notes generation more opaque.